utils/python/CIME/hist_utils.py

```diff
@@ -0,0 +1,175 @@
+"""
+Functions for actions pertaining to history files.
+"""
+
+from CIME.XML.standard_module_setup import *
+
+import logging, glob, os, shutil, re
+
+def _iter_model_file_substrs(case):
+
+    models = case.get_compset_components()
+    models.append('cpl')
+    for model in models:
+        yield model
+
+def _get_all_hist_files(testcase, model, suffix="", from_dir=os.getcwd()):
+    suffix = (".%s" % suffix) if suffix else ""
+    test_hists = glob.glob("%s/%s.%s*.h?.nc%s" % (from_dir, testcase, model, suffix))
+    test_hists.extend(glob.glob("%s/%s.%s*.h.nc%s" % (from_dir, testcase, model, suffix)))
```

Comment:
Can you add comments describing the types of files that each of these globs are matching? For example, "This glob matches multi-instance files". Without such comments, I'm worried that it will be hard to keep this function up to date as we change / add / delete certain history file naming patterns.